auth/services/irma/signer.go

```diff
@@ -138,6 +144,35 @@ func (v Service) SigningSessionStatus(sessionID string) (contract.SigningSession
 	return nil, services.ErrSessionNotFound
 }
 
+func (v Service) legalEntityFromContract(sic *SignedIrmaContract) (*did.DID, error) {
+	params := sic.contract.Params
+
+	q, err := v.VCResolver.Registry().QueryFor(concept.OrganizationConcept)
+	if err != nil {
+		return nil, err
+	}
+	q.AddClause(concept.Eq(concept.OrganizationName, params[contract.LegalEntityAttr]))
+	q.AddClause(concept.Eq(concept.OrganizationCity, params[contract.LegalEntityCityAttr]))
+
+	c, err := v.VCResolver.Search(q)
+	if err != nil {
+		return nil, err
+	}
+	if len(c) == 0 {
+		return nil, ErrLegalEntityNotFound
+	}
+
+	cs := make([]credential.BaseCredentialSubject, 0)
+	// can't go wrong: it comes from JSON
+	_ = c[0].UnmarshalCredentialSubject(&cs)
+
+	if len(cs) == 0 {
```

Comment:
Could there be multiple subjects?

Reply:
Currently this is not supported. The concept mapping also doesn't support it.